internal/factsengine/gatherers/saplocalhostresolver_test.go

```diff
@@ -0,0 +1,154 @@
+package gatherers_test
+
+import (
+	"errors"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+	mocks "github.com/trento-project/agent/internal/factsengine/gatherers/mocks"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/test/helpers"
+)
+
+type SapLocalhostResolverTestSuite struct {
+	suite.Suite
+	mockResolver *mocks.HostnameResolver
+	mockPinger   *mocks.HostPinger
+}
+
+func TestSapLocalhostResolverTestSuite(t *testing.T) {
+	suite.Run(t, new(SapLocalhostResolverTestSuite))
+}
+
+func (suite *SapLocalhostResolverTestSuite) SetupTest() {
+	suite.mockResolver = new(mocks.HostnameResolver)
+	suite.mockPinger = new(mocks.HostPinger)
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverSuccess() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	ascsProfileFile, _ := os.Open(helpers.GetFixturePath("gatherers/sap_profile.ascs"))
+	ascsProfileConcent, _ := io.ReadAll(ascsProfileFile)
+	err = afero.WriteFile(appFS, "/sapmnt/QAS/profile/QAS_ASCS00_sapqasas", ascsProfileConcent, 0644)
+	suite.NoError(err)
+
+	erProfileFile, _ := os.Open(helpers.GetFixturePath("gatherers/sap_profile.er.minimal"))
+	erProfileConcent, _ := io.ReadAll(erProfileFile)
+	err = afero.WriteFile(appFS, "/sapmnt/QAS/profile/NWP_ER10_sapnwper", erProfileConcent, 0644)
```

Comment:
This path is wrong. If should have `QAS` as `NWP` or viceversa. Not both. And better `ERS` than `ER`
By the way, I think we should have some test with a couple of SIDs and a couple of profiles